src/VecchiaMLE_defines.jl

```diff
@@ -65,18 +65,18 @@ mutable struct Diagnostics
     objective_value::Float64            # Optimal Objective value. 
     normed_constraint_value::Float64    # Optimal norm of constraint vector.
     normed_grad_value::Float64          # Optimal norm of gradient vector.
-    MadNLP_iterations::Integer          # Iterations for MadNLP to reach optimal.
-    mode::Integer                       # Operation mode: CPU or GPU
```

Comment:
@CalebDerrickson `Integer` is not concrete type. Julia can't determine statically if it's a Int32, Int64, etc...
It can only do that at runtime.